src/client/containers/OrderPage/OrderPage.Container.js

```diff
@@ -1,12 +1,121 @@
 import React from 'react';
+import ButtonComponent from '../../components/Button/Button.component';
+import ContactForm from '../../components/ContactForm/ContactForm.component';
+import DeliveryInfo from '../../components/DeliveryInfo/DeliveryInfo.component';
+import Loader from '../../components/Loader/Loader.component';
+import ShoppingItem from '../../components/ShoppingItem/ShoppingItem';
+import TotalPrice from '../../components/TotalPriceCard/TotalPriceCard.component';
+import { useParams, useHistory } from 'react-router-dom';
+import { useFetchApi } from '../../hooks/UseFetchApi';
+import { useFirebase } from '../../firebase/FirebaseContext';
+import { PropTypes } from 'prop-types';
 import './OrderPage.Style.css';
 
-const OrderPageContainer = () => {
+const OrderPageContainer = ({ isAuthenticated }) => {
+  const [items, setItems] = React.useState([]);
+  const [order, setOrder] = React.useState({});
+  const [user, setUser] = React.useState({});
+  const [total, setTotal] = React.useState(0);
+  const { id } = useParams();
+  const history = useHistory();
+  const { auth } = useFirebase();
+
+  const newItems = useFetchApi(`orders/${id}`);
+
+  React.useEffect(() => {
+    if (!newItems.isLoading) {
+      setItems(newItems.data.items);
+      setOrder(newItems.data.order);
+    }
+  }, [newItems]);
+
+  const userInfo = useFetchApi(`users/${order.userId}`);
+
+  React.useEffect(() => {
+    if (!userInfo.isLoading) {
```

Comment:
This part throws error in DeliveryInfo component when order is not loaded yet it is doing GET `api/users/undefined` and user is set to undefined.
You need to check also if order is loaded f.ex. `if (!userInfo.isLoading && !newItems.isLoading) `